snforge_std/src/trace.cairo

```diff
@@ -34,3 +34,160 @@ fn get_call_trace() -> CallTrace {
     let mut output = cheatcode::<'get_call_trace'>(array![].span());
     Serde::deserialize(ref output).unwrap()
 }
+
+use core::fmt::{Display, Formatter, Error, Debug};
+
+// Display
+
+impl DisplayEntryPointType of Display<EntryPointType> {
+    fn fmt(self: @EntryPointType, ref f: Formatter) -> Result<(), Error> {
+        let str: ByteArray = match self {
+            EntryPointType::Constructor => "Constructor",
+            EntryPointType::External => "External",
+            EntryPointType::L1Handler => "L1 Handler",
+        };
+        f.buffer.append(@str);
+        Result::Ok(())
+    }
+}
+
+impl DisplayCallType of Display<CallType> {
+    fn fmt(self: @CallType, ref f: Formatter) -> Result<(), Error> {
+        let str: ByteArray = match self {
+            CallType::Call => "Call",
+            CallType::Delegate => "Delegate",
+        };
+        f.buffer.append(@str);
+        Result::Ok(())
+    }
+}
+
+impl DisplayCallTrace of Display<CallTrace> {
+    fn fmt(self: @CallTrace, ref f: Formatter) -> Result<(), Error> {
+        Display::fmt(@IndentedCallTrace { struct_ref: self, base_indents: 0 }, ref f).unwrap();
+        Result::Ok(())
+    }
+}
+
+#[derive(Drop)]
+struct IndentedCallTrace {
+    struct_ref: @CallTrace,
+    base_indents: u8,
+}
+
+#[derive(Drop)]
+struct IndentedCallTraceArray {
+    struct_ref: @Array<CallTrace>,
+    base_indents: u8,
+}
+
+#[derive(Drop)]
+struct IndentedEntryPoint {
+    struct_ref: @CallEntryPoint,
+    base_indents: u8,
+}
+
+
+impl DisplayIndentedCallTrace of Display<IndentedCallTrace> {
+    fn fmt(self: @IndentedCallTrace, ref f: Formatter) -> Result<(), Error> {
+        Display::fmt(
+            @IndentedEntryPoint {
+                base_indents: *self.base_indents, struct_ref: *self.struct_ref.entry_point
+            },
+            ref f
+        )
+            .unwrap();
+        write!(f, "\n").unwrap();
+        write_indents_to_formatter(*self.base_indents, ref f);
+        write!(f, "Nested Calls: [").unwrap();
+        if (*self.struct_ref.nested_calls).len() > 0 {
+            write!(f, "\n").unwrap();
+            Display::fmt(
+                @IndentedCallTraceArray {
+                    base_indents: (*self.base_indents) + 1,
+                    struct_ref: *self.struct_ref.nested_calls
+                },
+                ref f
+            )
+                .unwrap();
+            write!(f, "\n").unwrap();
+            write_indents_to_formatter(*self.base_indents, ref f);
```

Comment:
Could be outside of the loop I think

Reply:
Actually no, if the array would be empty it'd result in `[    ]` instead of `[]`